### ISSUE TITLE
[api-extractor] Generating documentation for interface in bundled package

### ISSUE THREAD

**Issue by JustinBeckwith** [microsoft/rushstack#1933]
👋  Greetings folks!  Before all this, just wanted to say what a lovely toolchain you're building, and how jazzed we are to start using it.  

## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**
I am running `api-extractor` and `api-documenter`, and generating _a lot_ of markdown.  I have a dependency added as a `bundledPackage`, and many (but not all...) of the interfaces are being included.  For whatever reason, one interface isn't showing up as a hyperlink in the generated docs.  It is unclear to me if this is an issue in api-extractor or api-documenter.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**
I created a repository to demonstrate the issue here:
https://github.com/JustinBeckwith/extractor-issues

To reproduce:
```
git clone git@github.com:JustinBeckwith/extractor-issues.git
cd extractor-issues
npm install
npm run compile
npm run extract
npm run docs
```

The generated `d.ts` and rollup stuff will be in the `extractor` directory, and the generated markdown will be in the `docs` directory.  The doc in question is here:
https://github.com/JustinBeckwith/extractor-issues/blob/master/docs/vision.vision_v1.resource_files.annotate.md

![image](https://user-images.githubusercontent.com/534619/84616314-81814100-ae80-11ea-948d-ef2899d56df4.png)

`StreamMethodOptions` is an interface defined in the `googleapis-common` module, which is listed as an bundled package here:
https://github.com/JustinBeckwith/extractor-issues/blob/master/api-extractor.json#L65

And it's exported from `index.ts` here:
https://github.com/JustinBeckwith/extractor-issues/blob/master/index.ts#L62

The interface isn't anything special, and is defined here:
https://github.com/googleapis/nodejs-googleapis-common/blob/master/src/api.ts#L51

**What is the expected behavior?**
I would hope for the `StreamMethodOptions` link in the generated markdown to be a hyperlink pointing to the definition of the interface

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** api-extractor and maybe api-documenter
* **Tool Version:** 7.8.11 and 7.8.12
* **Node Version:** 14.4.0
  * **Is this a LTS version?** I  mean, not yet :) 
  * **Have you tested on a LTS version?** yes (12)
* **OS:** OSX

**Comment by octogonz** [comment]
This is a bug. It shows up in API Extractor's test cases, too: The bundled `Lib1Class` appears in [etc/test-outputs/bundledPackages/rollup.d.ts](https://github.com/microsoft/rushstack/blob/678a3a5393eeeec226858a400495f8769d430df4/build-tests/api-extractor-scenarios/etc/test-outputs/bundledPackages/rollup.d.ts#L7) but is missing from [the .api.json file](https://github.com/microsoft/rushstack/blob/678a3a5393eeeec226858a400495f8769d430df4/build-tests/api-extractor-scenarios/etc/test-outputs/bundledPackages/api-extractor-scenarios.api.json).

It was probably overlooked because `bundledPackages` is a relatively new feature that hasn't had a lot of usage yet. Let me see what's involved to fix it.

**Comment by octogonz** [comment]
Minor clarification: The `Lib1Class` entry is missing from that .api.json file only because it is not exported by the test project. If I export it, then it does appear in the .api.json file. But the hyperlink is missing in the generate documentation, which repros your issue.

The reason is that the `canonicalReference` is not considering the `bundledPackages` mapping.  

The `function f(arg1: Lib1Class, arg2: Lib2Class): void;` entry refers to the external package (which got bundled in this example):
```json
 "canonicalReference": "api-extractor-lib1-test!Lib1Class:class",
```
...whereas the `class Lib1Class` entry has been rewritten as part of the local project:
```json
 "canonicalReference": "api-extractor-scenarios!Lib1Class:class",
```
We need to make these the same. Given the semantics of `bundledPackages`, I'm thinking they should both be `"api-extractor-scenarios!Lib1Class:class"`.

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/1935

**Comment by JustinBeckwith** [comment]
Thank you so much for the fix 🥳

**Comment by octogonz** [comment]
This fix was published with API Extractor 7.8.12
